### PR TITLE
Add support for epicscorelibs and CI pypi publishing

### LINKED ISSUE #1: Broken API in EPICS 7
Hi Michael,

I've been looking at using the pythonIOC with a recent release of EPICS.

From EPICS 7.0.2.1 dbGetFieldType was removed, we can instead use dbGetFieldTypeString, which was introduced in R3.15.2. The change https://github.com/mfurseman/epicsdbbuilder/commit/16b28810ece67982e6528288c0f9e56720d74545 fixes EPICS 7, but will break compatibility with versions before R3.15.2. It seems some ugly changes would be required to maintain backwards compatibility.

### LINKED ISSUE #6: License information is missing
License information is not available. It is not clear whether the package can be used for given purpose. Would it be possible to add information about active license, please?

### PR DESCRIPTION
**Pull request by thomascobb**
Docs won't publish until we merge this to master, and a PyPI deployment won't go until we tag and push, but everything else should be done.

Fixes #6, supersedes #4

### CODE REVIEW

**Comment by thomascobb** [comment]
ok, ready for review now...

File: epicsdbbuilder/dbd.py [hunk 1]
```diff
@@ -134,9 +134,15 @@ def LoadDbdFile(dbdfile, on_use = None):
     mydbstatic.dbFreeEntry(entry)
 
 
-def InitialiseDbd(epics_base, host_arch = None):
+def InitialiseDbd(epics_base = None, host_arch = None):
     global _epics_base
+    if epics_base:
+        # Import from given location
+        mydbstatic.ImportFunctions(epics_base, host_arch)
     _epics_base = epics_base
```
Comment (Araneidae) [inline epicsdbbuilder/dbd.py]: This line is clearly out of place!

**Review by Araneidae — COMMENTED** [review]
Some changes under discussion

File: epicsdbbuilder/fanout.py [hunk 2]
```diff
@@ -19,9 +19,8 @@ def choplist(list, size):
 
 # Support routine to do the work of fanout generation common to fanout and
 # dfanout.
-def _fanout_helper(
-    fanout_name, link_list, fanout_size,
-    record_factory, field_name, fixup_link, firstargs, nextargs):
+def _fanout_helper(fanout_name, link_list, fanout_size, record_factory,
```
Comment (Araneidae) [inline epicsdbbuilder/fanout.py:22]: Don't indent to open bracket here!

File: epicsdbbuilder/recordbase.py [hunk 3]
```diff
@@ -119,9 +119,8 @@ def __init__(self, record, **fields):
         # Support the special 'address' field as an alias for either INP or
         # OUT, depending on which of those exists.  We only set up this field
         # if exactly one of INP or OUT is present as a valid field.
-        address = [field
-            for field in ['INP', 'OUT']
-            if self.ValidFieldName(field)]
+        address = [field for field in ['INP', 'OUT']
```
Comment (Araneidae) [inline epicsdbbuilder/recordbase.py:122]: Don't use brackets for indent start

File: Makefile [hunk 4]
```diff
@@ -1,31 +0,0 @@
-# Specify defaults for testing
```
Comment (Araneidae) [inline Makefile:1]: Think we'll still need this for dls-python 2.7 releases

**Comment by Araneidae** [comment]
Looks like there's a spurious `.coverage` database file introduced in commit b1a7ee264992bb49ea5a17202e0e8d717032fe3f

**Comment by Araneidae** [comment]
Can you please rebase out `.coverage` from commit b1a7ee264992bb49ea5a17202e0e8d717032fe3f, then we're good to go.

**Comment by thomascobb** [comment]
> Can you please rebase out `.coverage` from commit [b1a7ee2](https://github.com/dls-controls/epicsdbbuilder/commit/b1a7ee264992bb49ea5a17202e0e8d717032fe3f), then we're good to go.

done

**Comment by Araneidae** [comment]
Looks good, think I'm hapy

**Review by Araneidae — APPROVED** [review]

**Pull request merged**